File: tinta.php
<?php
// Dirección IP de la impresora Epson
$printer_ip = '192.168.0.242';  // Cambia esto por la IP de tu impresora

// Comunidad SNMP, generalmente es 'public'
$community = 'public';

// OIDs para los niveles de tinta (ajústalas si es necesario)
$oid_black_ink = '1.3.6.1.2.1.43.11.1.1.9.1.1';  // Nivel de tinta negra
$oid_cyan_ink = '1.3.6.1.2.1.43.11.1.1.9.1.2';   // Nivel de tinta cian
$oid_magenta_ink = '1.3.6.1.2.1.43.11.1.1.9.1.3'; // Nivel de tinta magenta
$oid_yellow_ink = '1.3.6.1.2.1.43.11.1.1.9.1.4';  // Nivel de tinta amarilla

// Función para obtener el nivel de tinta usando SNMP
function getInkLevel($ip, $community, $oid) {
    $result = @snmpget($ip, $community, $oid);
    if ($result !== false) {
        // Extraer solo los números del resultado SNMP
        $value = preg_replace('/[^0-9]/', '', $result);
        return intval($value);  // Devolver el valor como número
    } else {
        return "No se puede obtener el nivel de tinta";
    }
}

// Obtener niveles de tinta
$black_ink_level = getInkLevel($printer_ip, $community, $oid_black_ink);
$cyan_ink_level = getInkLevel($printer_ip, $community, $oid_cyan_ink);
$magenta_ink_level = getInkLevel($printer_ip, $community, $oid_magenta_ink);
$yellow_ink_level = getInkLevel($printer_ip, $community, $oid_yellow_ink);

// Mostrar niveles de tinta en formato JSON
header('Content-Type: application/json');
echo json_encode([
    'black' => $black_ink_level . '%',
    'cyan' => $cyan_ink_level . '%',
    'magenta' => $magenta_ink_level . '%',
    'yellow' => $yellow_ink_level . '%'
]);

File: tinta.py
import os
from pysnmp.hlapi import *
import smtplib
from email.mime.text import MIMEText
from datetime import datetime

# Función para obtener el nivel de tinta usando SNMP
def get_ink_level(ip, oid):
    iterator = getCmd(
        SnmpEngine(),
        CommunityData('public', mpModel=0),
        UdpTransportTarget((ip, 161)),
        ContextData(),
        ObjectType(ObjectIdentity(oid))
    )

    errorIndication, errorStatus, errorIndex, varBinds = next(iterator)

    if errorIndication:
        print(errorIndication)
        return None
    elif errorStatus:
        print('%s at %s' % (errorStatus.prettyPrint(),
                            errorIndex and varBinds[int(errorIndex) - 1][0] or '?'))
        return None
    else:
        for varBind in varBinds:
            # Imprime el resultado SNMP
            return int(varBind[1])  # Convertimos el valor a número entero

# OIDs para los niveles de tinta (ajústalas según el modelo de impresora)
oids = {
    'black': '1.3.6.1.2.1.43.11.1.1.9.1.1',
    'cyan': '1.3.6.1.2.1.43.11.1.1.9.1.2',
    'magenta': '1.3.6.1.2.1.43.11.1.1.9.1.3',
    'yellow': '1.3.6.1.2.1.43.11.1.1.9.1.4'
}

# IP de la impresora
printer_ip = input("Introduce la IP de la impresora: ")

# Obtener niveles de tinta
ink_levels = {color: get_ink_level(printer_ip, oid) for color, oid in oids.items()}

# Mostrar niveles de tinta en consola
print("Niveles de tinta:")
for color, level in ink_levels.items():
    print(f"{color.capitalize()}: {level}%")

# Guardar resultados en un archivo
with open('niveles_tinta.txt', 'a') as f:
    f.write(f"\n{datetime.now()} - IP: {printer_ip}\n")
    for color, level in ink_levels.items():
        f.write(f"{color.capitalize()}: {level}%\n")

# Enviar correo con los resultados (opcional)
def send_email_report(ink_levels, printer_ip):
    sender = 'tu_correo@example.com'
    recipient = 'destinatario@example.com'
    subject = 'Reporte de Niveles de Tinta'
    body = f"Niveles de tinta de la impresora ({printer_ip}):\n" + \
           '\n'.join([f"{color.capitalize()}: {level}%" for color, level in ink_levels.items()])

    msg = MIMEText(body)
    msg['From'] = sender
    msg['To'] = recipient
    msg['Subject'] = subject

    try:
        with smtplib.SMTP('smtp.example.com', 587) as server:
            server.starttls()
            server.login(sender, 'tu_contraseña')
            server.sendmail(sender, recipient, msg.as_string())
            print("Correo enviado exitosamente.")
    except Exception as e:
        print(f"Error al enviar el correo: {e}")

# Enviar reporte por correo (descomenta para habilitar)
# send_email_report(ink_levels, printer_ip)
